Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -131,6 +161,52 @@ Quality RawDataReaderCheck::check(std::map<std::string, std::shared_ptr<MonitorO
       else
         result = Quality::Good;
     }
+    if (mo->getName() == "inputRatio" || mo->getName() == "classRatio") {
```

Comment:
This block is a pretty difficult to understand. Could you perhaps add a few comments ? There are also quite many `if mo->getName() =="XXX"` that gets repeated. perhaps restructuring slightly would help.

Reply:
I added some comments and in the last commit I've rewritten the code. It is shorter now and hopefully also simpler to understand.